### PR TITLE
Add Mistral AI as a new provider in LLM Deployment

### LINKED ISSUE #10948: [FR] Add Mistral AI as a new LLM provider
### Willingness to contribute

No. I cannot contribute this feature at this time.

### Proposal Summary

I would like to know if we can have a way to support Mistral AI as a new provider in Mlflow LLM Deployment.

I've seen from the documentation [here](https://mlflow.org/docs/latest/llms/deployments/index.html#supported-provider-models) that mlflow supports many providers, and Mistral AI could be served in HuggingFace TGI. However, I consider if we can have a way to support it as Anthropic or OpenAI so that it can be used through Mlflow LLM Deployment's endpoint 'predict', that seems very simple to manage in the future.

Thank you in advance for your reply

### Motivation

> #### What is the use case for this feature?

> #### Why is this use case valuable to support for MLflow users in general?

> #### Why is this use case valuable to support for your project(s) or organization?

> #### Why is it currently difficult to achieve this use case?

### Details

_No response_

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by thnguyendn**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/thnguyendn/mlflow/pull/11020?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11020/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11020
```

</p>
</details>

### Related Issues/PRs

Resolve #10948

### What changes are proposed in this pull request?

Though MLFlow has already proposed Mistral AI as part of Hugging Face TGI, this PR is created to fill the need of having Mistral AI as a new LLM Provider, that could be easier to manage by the centralized endpoint of MLFlow LLM Deployment.

In terms of functionality, the new provider provides an endpoint for completion and another one for embedding, those which correspond to the Mistral AI completion (https://docs.mistral.ai/api/#operation/createChatCompletion) and embedding APIs (https://docs.mistral.ai/api/#operation/createEmbedding).

Technically, the changes in this Pull Request can be found in the following places:
- Creation of the MistralConfig in the config.py file
- Creation of the MistralProvider and MistralAdapter to adapt the request and the corresponding response from the Mistral APIs (mistral.py)
- Creation of a unit test to test MistralProvider (test_mistral.py)

### How is this PR tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [x] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [x] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by BenWilson2** [comment]
Hi @thnguyendn great stuff! Could you add an entry in the integration suite for this provider? https://github.com/mlflow/mlflow/blob/master/tests/gateway/test_integration.py

**Comment by thnguyendn** [comment]
> Hi @thnguyendn great stuff! Could you add an entry in the integration suite for this provider? https://github.com/mlflow/mlflow/blob/master/tests/gateway/test_integration.py

A new commit has been added to add entry in integration suite for the provider

**Comment by thnguyendn** [comment]
Hi @BenWilson2 , could you please review the PR?

**Review by BenWilson2 — APPROVED** [review]
Looks good to me with the added test! Thanks for the contribution :) 
Are you planning on a follow-up to add the examples and the docs updates?

**Comment by thnguyendn** [comment]
Hi @BenWilson2 , I've just updated docs and add examples for the provider.

Can you check it @BenWilson2 ?

Are you planning to add the PR in your next patch version 2.10.2 ?

File: docs/source/llms/deployments/index.rst [hunk 1]
```diff
@@ -682,6 +687,16 @@ To match your user's interaction and security access requirements, adjust the ``
 +----------------------------+----------+---------+-----------------------------------------------------------------------------------------------+
 
 
+Mistral
+++++++
```
Comment (BenWilson2) [inline docs/source/llms/deployments/index.rst:691]:
```suggestion
Mistral
+++++++
```

File: docs/source/llms/gateway/index.rst [hunk 2]
```diff
@@ -639,6 +644,16 @@ Top-level model configuration for AWS Bedrock routes must be one of the followin
 +--------------------------+----------+------------------------------+-------------------------------------------------------+
 
 
+Mistral
+++++++
```
Comment (BenWilson2) [inline docs/source/llms/gateway/index.rst:648]:
```suggestion
+++++++
```

**Pull request merged**